File: CS/Q144679/MyCheckedListBoxControl.cs
using DevExpress.XtraEditors;
using System.Windows.Forms;
using System.Reflection;
using DevExpress.XtraEditors.ViewInfo;

namespace DXSample {
    public class MyCheckedListBoxControl : CheckedListBoxControl {
        internal bool allowChangeCheck;
        public MyCheckedListBoxControl() : base() { }

        protected override ListBoxControlHandler CreateHandler() {
            return new MyCheckedListBoxControlHandler(this);
        }

        protected override void OnChangeCheck(int index) {
            if (!allowChangeCheck) return;
            base.OnChangeCheck(index);
        }
    }

    public class MyCheckedSingleSelectState : CheckedListBoxControlHandler.CheckedSingleSelectState {
        public MyCheckedSingleSelectState(MyCheckedListBoxControlHandler handler) : base(handler) { }

        public override void MouseUp(MouseEventArgs e) {
            FieldInfo fi = typeof(BaseStyleControl).GetField("viewInfo", 
                BindingFlags.Instance | BindingFlags.NonPublic);
            CheckedListBoxViewInfo viewInfo = (CheckedListBoxViewInfo)fi.GetValue(ListBox);
            CheckedListBoxViewInfo.CheckedItemInfo itemInfo = (CheckedListBoxViewInfo.CheckedItemInfo)viewInfo.GetItemInfoByPoint(e.Location);
            if (itemInfo != null && itemInfo.CheckArgs.GlyphRect.Contains(e.Location))
                ((MyCheckedListBoxControl)ListBox).allowChangeCheck = true;
            base.MouseUp(e);
            ((MyCheckedListBoxControl)ListBox).allowChangeCheck = false;
        }
    }

    public class MyCheckedListBoxControlHandler : CheckedListBoxControlHandler {
        public MyCheckedListBoxControlHandler(MyCheckedListBoxControl control) : base(control) { }

        protected override ListBoxControlState CreateState(HandlerState state) {
            if (state == HandlerState.Unselectable)
                return base.CreateState(state);
            return new MyCheckedSingleSelectState(this);
        }
    }
}

File: VB/Q144679/MyCheckedListBoxControl.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports System.Windows.Forms
Imports System.Reflection
Imports DevExpress.XtraEditors.ViewInfo

Namespace DXSample
	Public Class MyCheckedListBoxControl
		Inherits CheckedListBoxControl
		Friend allowChangeCheck As Boolean
		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides Function CreateHandler() As ListBoxControlHandler
			Return New MyCheckedListBoxControlHandler(Me)
		End Function

		Protected Overrides Sub OnChangeCheck(ByVal index As Integer)
			If (Not allowChangeCheck) Then
				Return
			End If
			MyBase.OnChangeCheck(index)
		End Sub
	End Class

	Public Class MyCheckedSingleSelectState
		Inherits CheckedListBoxControlHandler.CheckedSingleSelectState
		Public Sub New(ByVal handler As MyCheckedListBoxControlHandler)
			MyBase.New(handler)
		End Sub

		Public Overrides Sub MouseUp(ByVal e As MouseEventArgs)
			Dim fi As FieldInfo = GetType(BaseStyleControl).GetField("viewInfo", BindingFlags.Instance Or BindingFlags.NonPublic)
			Dim viewInfo As CheckedListBoxViewInfo = CType(fi.GetValue(ListBox), CheckedListBoxViewInfo)
			Dim itemInfo As CheckedListBoxViewInfo.CheckedItemInfo = CType(viewInfo.GetItemInfoByPoint(e.Location), CheckedListBoxViewInfo.CheckedItemInfo)
			If itemInfo IsNot Nothing AndAlso itemInfo.CheckArgs.GlyphRect.Contains(e.Location) Then
				CType(ListBox, MyCheckedListBoxControl).allowChangeCheck = True
			End If
			MyBase.MouseUp(e)
			CType(ListBox, MyCheckedListBoxControl).allowChangeCheck = False
		End Sub
	End Class

	Public Class MyCheckedListBoxControlHandler
		Inherits CheckedListBoxControlHandler
		Public Sub New(ByVal control As MyCheckedListBoxControl)
			MyBase.New(control)
		End Sub

		Protected Overrides Function CreateState(ByVal state As HandlerState) As ListBoxControlState
            If state = HandlerState.Unselectable Then
                Return MyBase.CreateState(state)
            End If
			Return New MyCheckedSingleSelectState(Me)
		End Function
	End Class
End Namespace